### ISSUE TITLE
Add more options to addBearerAuth() method

### ISSUE THREAD

**Issue by patrickhousley** [nestjs/swagger#83]
Document configuration:
```ts
export const swaggerOptions = new DocumentBuilder()
  .setTitle('Blog Application')
  .setDescription('APIs for the example blog application.')
  .setVersion('1.0.0')
  .setHost('localhost:3000')
  .setBasePath('/')
  .setSchemes('http')
  .setExternalDoc('For more information', 'http://swagger.io')
  .addTag('blog', 'application purpose')
  .addTag('nestjs', 'framework')
  .addBearerAuth('Authorization', 'header', 'apiKey')
  .addOAuth2(
    'password',
    'http://localhost:3000/login',
    'http://localhost:3000/token',
    {
      test: 'test'
    }
  )
  .build();
```
API:
```ts
@Put('comments/:commentId')
@ApiBearerAuth()
@ApiOAuth2Auth(['test'])
public async update(
    @Body() body: UpdateCommentRequest,
    @Param('commentId') comment: string,
    @Query('testQuery') testQuery: string,
    @Headers('testHeader') testHeader: string
) {}
```
Generated Swagger:
```json
"/entries/{entryId}/comments/{commentId}": {
  "put": {
    "parameters": [{
        "type": "string",
        "name": "testHeader",
        "required": true,
        "in": "header"
      },
      {
        "type": "string",
        "name": "testQuery",
        "required": true,
        "in": "query"
      },
      {
        "type": "string",
        "name": "commentId",
        "required": true,
        "in": "path"
      },
      {
        "type": class UpdateCommentRequest {},
        "name": "UpdateCommentRequest",
        "required": true,
        "in": "body",
        "schema": {
          "$ref": "#/definitions/UpdateCommentRequest"
        }
      }
    ],
    "responses": {},
    "security": {
      "security": [{
          "bearer": []
        },
        {
          "oauth2": [
            "test"
          ]
        }
      ]
    },
    "produces": [
      "application/json"
    ],
    "consumes": [
      "application/json"
    ]
  }
}
```

The security section is not formatted correctly. It should be:
```json
"/entries/{entryId}/comments/{commentId}": {
  "put": {
    "parameters": [{
        "type": "string",
        "name": "testHeader",
        "required": true,
        "in": "header"
      },
      {
        "type": "string",
        "name": "testQuery",
        "required": true,
        "in": "query"
      },
      {
        "type": "string",
        "name": "commentId",
        "required": true,
        "in": "path"
      },
      {
        "type": class UpdateCommentRequest {},
        "name": "UpdateCommentRequest",
        "required": true,
        "in": "body",
        "schema": {
          "$ref": "#/definitions/UpdateCommentRequest"
        }
      }
    ],
    "responses": {},
    "security": [
      { "bearer": [] },
      {
        "oauth2": [
          "test"
        ]
      }
    ],
    "produces": [
      "application/json"
    ],
    "consumes": [
      "application/json"
    ]
  }
}
```

Additionally, the `addBearerAuth` should accept two name properties: one for the name of the security scheme and one for the name of the property. the `@ApiBearerAuth` decorator should take the name of the security scheme provided to the `addBearerAuth` method. For example:

```ts
.addBearerAuth('domain', 'domain', 'query', 'apiKey')
.addBearerAuth('apiToken', 'token', 'query', 'apiKey')
```

This describes two api keys necessary for authentication. The names of the security schemes are `domain` and `apiToken` and their values are provides as query params `domain` and `token` respectively.

```ts
@ApiBearerAuth('domain')
@ApiBearerAuth('apiToken')
```

This describes an API that requires both the `domain` and `apiToken` security schemes for authentication.

**Comment by patrickhousley** [comment]
The `type` for the body parameter is also a little strange. If the `@Body` parameter has an interface or any type, the Swagger UI will actually throw an error `Uncaught SyntaxError: Unexpected identifier`.

**Comment by kamilmysliwiec** [comment]
```
The type for the body parameter is also a little strange. If the @Body parameter has an interface or any type, the Swagger UI will actually throw an error Uncaught SyntaxError: Unexpected identifier.
```
Fixed in `2.0.1`. Will take about the rest soon

**Comment by kamilmysliwiec** [comment]
4.0.0 has been published
